gravity-info-server/src/transactions.rs

```diff
@@ -0,0 +1,377 @@
+            };
+            let tx_body: TxBody = decode_any(body_any).unwrap();
+
+            let mut has_msg_send_to_eth = false;
+            let mut has_msg_ibc_transfer = false;
+
+            // tx sorting
+            for message in tx_body.messages {
+                if message.type_url == "/gravity.v1.MsgSendToEth" {
+                    has_msg_send_to_eth = true;
+                    msg_counter += 1;
+
+                    let msg_send_to_eth_any = prost_types::Any {
+                        type_url: "/gravity.v1.MsgSendToEth".to_string(),
+                        value: message.value,
+                    };
+                    let msg_send_to_eth: Result<MsgSendToEth, _> = decode_any(msg_send_to_eth_any);
+
+                    if let Ok(msg_send_to_eth) = msg_send_to_eth {
+                        let custom_msg_send_to_eth = CustomMsgSendToEth::from(&msg_send_to_eth);
+                        let key = format!("msgSendToEth_{}", tx_hash);
+                        save_msg_send_to_eth(db, &key, &custom_msg_send_to_eth);
+                    }
+                } else if message.type_url == "/ibc.applications.transfer.v1.MsgTransfer" {
+                    has_msg_ibc_transfer = true;
+                    msg_counter += 1;
+
+                    let msg_ibc_transfer_any = prost_types::Any {
+                        type_url: "/ibc.applications.transfer.v1.MsgTransfer".to_string(),
+                        value: message.value,
+                    };
+                    let msg_ibc_transfer: Result<MsgTransfer, _> = decode_any(msg_ibc_transfer_any);
+
+                    if let Ok(msg_ibc_transfer) = msg_ibc_transfer {
+                        let custom_ibc_transfer = CustomMsgTransfer::from(&msg_ibc_transfer);
+                        let key = format!("msgIbcTransfer_{}", tx_hash);
+                        save_msg_ibc_transfer(db, &key, &custom_ibc_transfer);
+                    }
+                }
+            }
+
+            if has_msg_send_to_eth {
+                tx_counter += 1;
+                send_eth_counter += 1;
+
+            }
+            if has_msg_ibc_transfer {
+                tx_counter += 1;
+                ibc_transfer_counter += 1;
+            }
+        }
+        current_block += 1;
+    }
+    let mut c = COUNTER.write().unwrap();
+    c.blocks += blocks_len;
+    c.transactions += tx_counter;
+    c.msgs += msg_counter;
+    c.ibc_msgs += ibc_transfer_counter;
+    c.send_eth_msgs += send_eth_counter;
+    info!(
+        "Parsed {} blocks, {} transactions, {} ibc messages, {} send to eth messages",
+        blocks_len, tx_counter, ibc_transfer_counter, send_eth_counter
+    );
+}
+
+pub fn transactions(api_db: web::Data<Arc<DB>>, db: Arc<DB>, db_options: &Options) -> tokio::task::JoinHandle<()> {
+    info!("Started downloading & parsing transactions");
+    tokio::spawn(async move {
+    let contact = Contact::new("http://gravity-grpc.polkachu.com:14290", TIMEOUT, "gravity")
```

Comment:
this should probably be a constant at the top of the file, also we already have the url in the main.rs we can re-use